anitya/tests/test_flask_api_v2.py

```diff
@@ -587,6 +587,66 @@ def test_list_projects(self):
 
         self.assertEqual(data, exp)
 
+    def test_list_projects_with_same_name(self):
+        """Assert two projects with the same name are sorted by the ecosystem."""
+        project1 = models.Project(
+            name="zlib",
+            homepage="https://hackage.haskell.org/package/zlib",
+            backend='GitHub',
+            ecosystem_name="https://hackage.haskell.org/package/zlib",
+        )
+        project2 = models.Project(
+            name="zlib",
+            homepage="http://www.zlib.net/",
+            backend="custom",
+            regex="DEFAULT",
+            ecosystem_name="http://www.zlib.net/",
+        )
+        session = Session()
+        session.add_all([project1, project2])
+        session.commit()
+
+        output = self.app.get('/api/v2/projects/')
+        self.assertEqual(output.status_code, 200)
+        data = _read_json(output)
+
+        for item in data['items']:
+            del item['created_on']
+            del item['updated_on']
+
+        exp = {
+            'page': 1,
+            'items_per_page': 25,
+            'total_items': 2,
+            'items': [
+                {
+                    "id": 2,
+                    "backend": "custom",
+                    "homepage": "http://www.zlib.net/",
+                    "name": "zlib",
+                    "regex": "DEFAULT",
+                    "version": None,
+                    "version_url": None,
+                    "versions": [],
+                    "ecosystem": "http://www.zlib.net/",
+                },
+                {
+                    "id": 1,
+                    "backend": "GitHub",
+                    "homepage": "https://hackage.haskell.org/package/zlib",
+                    "name": "zlib",
+                    "regex": None,
+                    "version": None,
+                    "version_url": None,
+                    "versions": [],
+                    "ecosystem": "https://hackage.haskell.org/package/zlib",
+                }
+            ]
+        }
+
+        self.maxDiff = None
```

Comment:
You probably want to set this elsewhere :)

Reply:
@pypingou Did you have a particular place in mind? I don't really mind it where it is. Maybe at the top of the test would be nicer, but that's a very minor nitpick.

Reply:
Thank you both for your review.

Indeed there are other test cases in the same file using it at the top of the test:
https://github.com/release-monitoring/anitya/blob/3433f08242d2ed542f9ba0d61614659dc8152a9d/anitya/tests/test_flask_api_v2.py#L357-L359

I will do the same.

I will also move `session = Session()` to be consistent with
https://github.com/release-monitoring/anitya/blob/3433f08242d2ed542f9ba0d61614659dc8152a9d/anitya/tests/test_flask_api_v2.py#L512-L516

Reply:
I'd even put it at the top of the class since it may influence other tests run after this one

Reply:
I'm not sure what you mean here, but if you're concerned setting it in an individual test affects any test run afterwards, that doesn't seem to be the case. That would be a very weird behavior.